File: src/ipc/viewModes.ts
import { invoke } from '@tauri-apps/api/core';

/** 同步原生菜单「专注 / 打字机」勾选状态 */
export async function setViewModeMenuChecked(
  focus: boolean,
  typewriter: boolean,
): Promise<void> {
  await invoke('set_view_mode_menu_checked', { focus, typewriter });
}

File: src-tauri/src/commands/view_modes.rs
use tauri::{AppHandle, Wry};

use crate::menu;

#[tauri::command]
pub fn set_view_mode_menu_checked(
    app: AppHandle<Wry>,
    focus: bool,
    typewriter: bool,
) -> Result<(), String> {
    menu::set_view_mode_checked(&app, focus, typewriter);
    Ok(())
}
